src/Workspaces/Core/Portable/SemanticModelReuse/AbstractSemanticModelReuseLanguageService.cs

```diff
@@ -53,13 +50,9 @@ internal abstract class AbstractSemanticModelReuseLanguageService<
 
                     try
                     {
-                        throw new InvalidOperationException(
-                            $@"Syntax trees should have been equivalent.
----
-{previousSyntaxTree.GetText(CancellationToken.None)}
----
-{currentSyntaxTree.GetText(CancellationToken.None)}
----");
+                        // We can't report the actual contents of the syntax tree since it may contain private
+                        // user information.
+                        throw new InvalidOperationException("Syntax trees should have been equivalent.");
```

Comment:
this actually defeats the purpose of this exception entirely, which is so we can figure out what went wrong.   It will also not actually change anything as the dumps will still contain this info, just in a *much* more painful manner to determine (often using windbg) in case the stack has moved elsewhere.

Reply:
A better pattern here may be to create a custom exception type here which has a generic message but also holds as properties the before/after. We've done that in other places and it can make debugging a bit easier, especially if the trees are different only by something hard to see (whitespace, for example.)

Reply:
Instead would hashing the content be the way to go?

Reply:
i'm totally ok with that @jasonmalinowski

Reply:
@arkalyanms In this case no, since any failure here means we're in a code construct that either has a compiler bug or broken code. The hash wouldn't be useful for diagnosing.